dask_jobqueue/tests/test_htcondor.py

```diff
@@ -75,6 +83,92 @@ def test_basic(loop):
                 assert time() < start + QUEUE_WAIT
 
 
+@pytest.mark.env("htcondor")
+def test_extra_args_broken_submit(loop):
+    with HTCondorCluster(
+        cores=1,
+        memory="100MB",
+        disk="100MB",
+        loop=loop,
+        submit_command_extra=["-name", "wrong.docker"],
+        cancel_command_extra=["-name", "wrong.docker"],
+    ) as cluster:
+        with pytest.raises(ValueError, match="Could not parse job id"):
+            cluster._dummy_job.start()
+
+
+@pytest.mark.env("htcondor")
+def test_extra_args_working_submit(loop):
+    schedd_name = Job._call(
+        ["condor_status", "-schedd", "-af", "Name", "-limit", "1"]
+    ).strip()
+    with HTCondorCluster(
+        cores=1,
+        memory="100MB",
+        disk="100MB",
+        loop=loop,
+        submit_command_extra=["-name", schedd_name],
+        cancel_command_extra=["-name", schedd_name],
+    ) as cluster:
+        with Client(cluster) as client:
+
+            cluster.scale(2)
+
+            start = time()
+            client.wait_for_workers(2)
+
+            future = client.submit(lambda x: x + 1, 10)
+            assert future.result(QUEUE_WAIT) == 11
+
+            workers = list(client.scheduler_info()["workers"].values())
+            w = workers[0]
+            assert w["memory_limit"] == 1e8
+            assert w["nthreads"] == 1
+
+            cluster.scale(0)
+
+            start = time()
+            while client.scheduler_info()["workers"]:
+                sleep(0.100)
+                assert time() < start + QUEUE_WAIT
+
+
+@pytest.mark.env("htcondor")
+def test_extra_args_broken_cancel(loop):
+    schedd_name = Job._call(
+        ["condor_status", "-schedd", "-af", "Name", "-limit", "1"]
+    ).strip()
+    with HTCondorCluster(
+        cores=1,
+        memory="100MB",
+        disk="100MB",
+        loop=loop,
+        submit_command_extra=["-name", schedd_name],
+        cancel_command_extra=["-name", "wrong.docker"],
+    ) as cluster:
+        with Client(cluster) as client:
+
+            cluster.scale(2)
+
+            client.wait_for_workers(2)
+
+            cluster.scale(0)
+
+            start = time()
+            while client.scheduler_info()["workers"]:
```

Comment:
one last thing: does this loop really sense? I first thought I understand but now I am unsure: It checks until the time out that the workers remain in the queue? But why does it assert `not workers`?

Reply:
You're right, I got the test backwards.